shared/errors.go

```diff
@@ -0,0 +1,7 @@
+package shared
```

Comment:
odd package name, why not keep it at top level?

Reply:
ah like just put this under package main too?

Reply:
yes

Reply:
oh so I think the reason why I made this another package is because both main and the gcp storage fuctions use it - if I put it in main it would be a cyclic import?

Reply:
hm, `gcp` can depend on `main` but not the other way around.

Reply:
ah I see ok let me try that